src/lib/Libifl/pbsD_submit_resv.c

```diff
@@ -70,9 +70,64 @@ __pbs_submit_resv(int c, struct attropl *attrib, char *extend)
 	struct attropl *pal;
 	int rc;
 	char *ret;
+	char job_id_resv[PBS_MAXJOBNAME + 1];
+	int msvr = multi_svr_op(c);
 
-	for (pal = attrib; pal; pal = pal->next)
+	job_id_resv[0] = '\0';
+	for (pal = attrib; pal; pal = pal->next) {
 		pal->op = SET;		/* force operator to SET */
+		if (msvr) {
+			if (strcmp(pal->name, ATTR_convert) == 0 || strcmp(pal->name, ATTR_resv_job) == 0) {
+				strcpy(job_id_resv, pal->value);
+			}
+		}
+	}
+
+	if (msvr) {
+		/* Job specific reservation */
+		if (job_id_resv[0] != '\0') {
+			/* Do a job stat to find out the server_instance_fd of
+			* the server instance where the job resides
+			*/
+			struct attrl *attr;
+			struct batch_status *ss = NULL;
+			char *svr_inst_id = NULL;
+			static struct attrl attribs[] = {
+				{	NULL,
+					ATTR_server_inst_id,
+					NULL,
+					"",
+					SET
+				}	
+			};
+
+			ss = pbs_statjob(c, job_id_resv, attribs, NULL);
+			if (ss == NULL)
+				return NULL;
+
+			if (ss != NULL) {
+				for (attr = ss->attribs; attr != NULL; attr = attr->next) {
+					if (strcmp(attr->name, ATTR_server_inst_id) == 0) { 
+						svr_inst_id = strdup(attr->value);
+						if (svr_inst_id == NULL) {
+							pbs_statfree(ss);
+							return NULL;
+						}
+						break;
+					}
+				}
+			}    
+			c = get_svr_inst_fd(c, svr_inst_id);
+			free(svr_inst_id);
+			pbs_statfree(ss);
```

Comment:
can you please put this in a separate function? something like get_resvjob_svr_inst_id()? it'll make this function look much cleaner.

Reply:
I have moved the above code into the function get_job_svr_inst_fd(). It is actually getting called only once at the moment. But since it can be used by any client, I have moved this to ifl_utils. c